packages/flutter/test/widgets/editable_text_test.dart

```diff
@@ -5183,6 +5183,52 @@ void main() {
     // toolbar. Until we change that, this test should remain skipped.
   }, skip: kIsWeb); // [intended]
 
+
+  testWidgets('text selection handle visibility for long text', (WidgetTester tester) async {
+    // long text which is scrollable based on given box size
+    const String testText =
+        'Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.';
+    final TextEditingController controller =
+        TextEditingController(text: testText);
+    final ScrollController scrollController = ScrollController();
+
+    await tester.pumpWidget(MaterialApp(
+      home: Align(
+        alignment: Alignment.topLeft,
+        child: SizedBox(
+          width: 100,
+          height: 100,
+          child: SingleChildScrollView(
+            controller: scrollController,
+            child: EditableText(
+              controller: controller,
+              showSelectionHandles: true,
+              focusNode: FocusNode(),
+              style: Typography.material2018().black.subtitle1!,
+              cursorColor: Colors.blue,
+              backgroundCursorColor: Colors.grey,
+              selectionControls: materialTextSelectionControls,
+              readOnly: true,
```

Comment:
Do you mind adding a `readOnly = false` version of the same test? I think the fix still applies on editable text fields too.

Reply:
absolutely, same is valid for `readOnly = false`, I remove this line to not confuse. I had an read-only use case